Fix RAM display on non-Linux hosts

Rycieos opened the pull request:
A number of small bugs that worked together to make Linux RAM display correct, but MacOS and BSD off by a factor of 1024.

Fixes #787

Note that this was not caught by the tests, as the tests only check what `__lp_ram_bytes()` returns, not what `_lp_ram()` returns. Would be good to have that added @nojhan if you want to do that. Specifically the values of `$lp_ram` and `$lp_ram_human`.

tomhoover commented:
I reset `LP_ENABLE_RAM=1` with your proposed fix. I'm now seeing `M32.00GiB` in my prompt, even though my MacBook Pro has only 16GB.

tomhoover commented:
More info:

Pressing `return` multiple times resulted in the following:
```
⏚ M225.17GiB [:~] 1 %
⏚ M47.56GiB [:~] 1 %
⏚ M62.39GiB [:~] 1 %
⏚ M256.98GiB [:~] 1 %
⏚ M218.29GiB [:~] 1 %
⏚ M198.43GiB [:~] 1 %
⏚ M149.60GiB [:~] 1 %
⏚ M134.96GiB [:~] 1 %
⏚ M128.87GiB [:~] 1 %
⏚ M122.40GiB [:~] 1 %
⏚ M110.14GiB [:~] 1 %
⏚ M114.84GiB [:~] 1 %
⏚ M111.43GiB [:~] 1 %
⏚ M108.31GiB [:~] 1 %
```
I'll be happy to run any tests you'd like.

nojhan commented:
I added tests on `lp_ram_*` and (hopefully) fixed the MacOS mem counter. On my mac it seems to be (more or less) consistent with the activity monitor.

Rycieos commented:
Looks like your fix was a step in the right direction. I found an issue with adding the two fields, which I fixed. I also fixed the test case. I added my own Mac as a test case, and fixed an issue I discovered on Cygwin.

nojhan reviewed (approved):
Seems good, works well on my setups.
